pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/kafka/AvroRecordToPinotRowGenerator.java

```diff
@@ -49,13 +48,40 @@ public GenericRow transform(@Nonnull GenericData.Record from, @Nonnull GenericRo
           fieldSpec.getFieldType() == FieldSpec.FieldType.TIME ? _incomingTimeFieldSpec : fieldSpec;
       String fieldName = incomingFieldSpec.getName();
       Object avroValue = from.get(fieldName);
-      if (incomingFieldSpec.isSingleValueField()) {
+      //Handle MAP types
+      if (fieldName.toUpperCase().endsWith("__KEYS")) {
```

Comment:
Hmm, this doesn't seem like the input field is a map? Also, relying on suffix seems quite brittle. What happens if there are multiple columns with these suffixes?

Reply:
I will update the description. But to answer your question, that should not be a problem. If you want to ingest a map field by name myMapColumn. The schema should contain myMapColumn__KEYS and myMapColumn__VALUES fields.

Reply:
I don't fully like the idea of relying of name suffix. But if this is an intermediate step that will get cleaned up later, I am fine with that.

Also, IIRC, we sort the values of MV column before storing. If so, that would probably break the key-value mapping between the two MV columns.

Reply:
yes, I don't like it either. Let me know if you can think of a better way to get there.

Reply:
I changed that to not sort it. It was not adding any value because we don't rely on the sort order.

Reply:
👍